### DISCUSSION TITLE
v2.6.100

### DISCUSSION THREAD

**Discussion by 2881099** [dotnetcore/FreeSql#867] [General]
- 修复 fix sqlite AddMinutes seconds->minutes #774
- 修复 Update操作的时候CanUpdate=false 未生效 #803
- 修复 Sqlite where.And(x => x.PublishTime.Year == 2021) 查询问题 #804
- 修复 Oracle IncludeMany IN 元素超过 500 数目的问题；#843
- 修复 MERGE INTO 别名与 SQL 关键字冲突的 bug；#816
- 优化 将Freesql的dynamicfilterinfo标记为[serialable] #802
- 增加 支持符号调试 #679
- 增加 IUnitOfWork.States 自定义状态管理，便于扩展；

-----------------------

## v2.5.200

- 修复 Repository/DbContext 批量修改可能无效的 bug；#709
- 修复 fix ExecuteMySqlBulkCopyAsync .net core 3.1 Method not found #783 需升级 v2.5.201

<hr /><em>This discussion was created from the release <a href='https://github.com/dotnetcore/FreeSql/releases/tag/v2.6.100'>v2.6.100</a>.</em>

**Comment by 2881099** [comment]
## IUnitOfWork.States 

使用者决定，用于扩展工作单元。

场景：跨方法失掉上下文的时候

比如给 IUnitOfWork 增加扩展方法的时候，下次再使用 uow，还知道存的什么内容

```c#
public static IBaseRepository<T> Crud<T>(this IUnitOfWork uow)
{
     if (uow.States.TryGetValue(typeof(T).FullName, out var repo)) return repo;
     repo = uow.Orm.GetRepository<T>();
     repo.UnitOfWork = uow;
     uow.States.Add(typeof(T).FullName, repo);
     return repo;
}

//使用 States 属性，工作单元比较好扩展
uow.Curd<Topic>().Insert(xxx);
uow.Curd<Topic>().Update(xxx);
```

**Comment by youtao** [comment]
你很棒
